includes/object-cache.php

```diff
@@ -566,13 +566,14 @@ public function __construct($fail_gracefully = true)
                 $this->redis_client .= sprintf(' (v%s)', Predis\Client::VERSION);
             }
 
-            if (defined('WP_REDIS_CLUSTER')) {
-                $this->redis->ping(current(array_values(WP_REDIS_CLUSTER)));
+            if ((strcasecmp('predis', $client) === 0) && defined('WP_REDIS_CLUSTER')) {
+                // Redis Cluster does not support the Ping command
+                //$this->redis->ping(current(array_values(WP_REDIS_CLUSTER)));
             } else {
                 $this->redis->ping();
             }
 
-            if ( ! isset( $options['replication'] ) || ! $options['replication'] ) {
+            if ( ((strcasecmp('predis', $client) === 0) && !defined('WP_REDIS_CLUSTER')) && (! isset( $options['replication'] ) || ! $options['replication']) ) {
```

Comment:
These should be extracted into variables, such as `$usesReplication` and `$onPredisCluster`.